metrics_utility/test/renewal_guidance/test_renew_guidance_dataframe.py

```diff
@@ -0,0 +1,90 @@
+import random
+import uuid
+
+from datetime import datetime, timezone
+
+import pandas as pd
+
+
+def generate_renewal_guidance_dataframe():
+    """
+    Generates a pandas DataFrame with specific, hardcoded mock renewal guidance
+    report data, matching the exact column names from the SQL query output.
+    """
+
+    # Helper to generate common datetime objects
+    first_auto_dt = datetime(2025, 5, 7, 15, 40, 6, tzinfo=timezone.utc)
+    last_auto_dt = datetime(2025, 5, 22, 19, 45, 11, tzinfo=timezone.utc)
+    # The last_deleted_dt should be an ISO8601 string or None, matching the DB
+    last_deleted_dt = (datetime(2025, 5, 20, 10, 0, 0, tzinfo=timezone.utc)).isoformat(timespec='seconds')
```

Comment:
see https://github.com/ansible/metrics-utility/blob/devel/metrics_utility/automation_controller_billing/dataframe_engine/db_dataframe_host_metric.py#L29

so, no `isoformat` here (unless you want to call `pd.to_datetime` on the whole thing afterwards), the datetime itself should work for this one